# Abstract {.unnumbered}

Understanding metabolic pathways in gut microbes is crucial for advancing microbiome research, yet identifying specific metabolic intermediates remains challenging due to the complexity of LC-MS data analysis. IsoPairFinder is a novel tool designed to streamline Stable Isotope Tracing (SIT) metabolomics from gene-edited microbes, enabling efficient identification of pathway intermediates and gene functions. IsoPairFinder offers an end-to-end workflow that: (1) Detects differential ion signals resulting from gene mutations; (2) Consolidates redundant LC-MS features (isotopes, adducts, fragments); (3) Pairs ¹²C/¹³C features to highlight candidate intermediates. Compatible with popular metabolomics data processing tools, IsoPairFinder empowers researchers to rapidly generate actionable biological insights. This tutorial provides step-by-step guidance for using **IsoPairFinder**. 


The ways to access *IsoPairFinder* are listed below:

| Access Method    | Type | Link | Document |
|------------------ |------------------|---------|---------|
| R Package        | Command line tools | [Link](https://github.com/DoddLab/IsoPairFinder) | [Documentation](https://doddlab.github.io/IsoPairFinder_Tutorials/) |
| Web Server       | Graphic interface within GNPS2 ecosystem | [Link](https://gnps2.org/workflowinput?workflowname=isopairfinder_nextflow_workflow) | [Documentation](https://wang-bioinformatics-lab.github.io/GNPS2_Documentation/isopairfinder/) |

<br>

If you used **IsoPairFinder** in your research, please cite the following paper:

- Zhou, Z., et al., IsoPairFinder: A tool for biochemical pathway discovery using stable isotope tracing metabolomics, In preparation, **2025**.

For questions, contact **Zhiwei Zhou** (Stanford University).